docs/spec_draft.md

```diff
@@ -2279,6 +2280,85 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.triangular_solve
+
+### Semantics
+
+Solves systems of linear equations of a batch of matrices with lower or upper
+triangular coefficients by forward- or back-substitution.
+
+More formally, given `a` and `b`, `result[i0, ..., iR-3, :, :]` is the solution
+to `op(a[i0, ..., iR-3, :, :]) * x = b[i0, ..., iR-3, :, :]` when `left_side` is
+`true` or `x * op(a[i0, ..., iR-3, :, :]) = b[i0, ..., iR-3, :, :]` when
+`left_side` is `false`, solving for the variable `x` where `op(a)` is either
+`op(a) = a`, or `op(a) = Transpose(a)`, or `op(a) = Conj(Transpose(a))`.
```

Comment:
How about "where `op(a)` is determined by `transpose_a` as described below"  to avoid duplication and introducing meta-functions (e.g. `Transpose` which is inconsistent with `transpose`)?

Reply:
Makes sense. I moved the description for `transpose_a` for this to flow better.